File: CS/ThreadedSkinning/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.LookAndFeel;

namespace ThreadedSkinning {
    public partial class Form2 : XtraForm {
        public Form2() {
            InitializeComponent();
        }

        public void MainLookAndFeel_StyleChanged(object sender, EventArgs e) {
            if(this.InvokeRequired) {
                this.BeginInvoke(new EventHandler(MainLookAndFeel_StyleChanged), sender, e);
                return;
            }

            UserLookAndFeel.Default.Assign((UserLookAndFeel)sender);
        }

    }
}

File: VB/ThreadedSkinning/Form2.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.LookAndFeel

Namespace ThreadedSkinning
	Partial Public Class Form2
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub MainLookAndFeel_StyleChanged(ByVal sender As Object, ByVal e As EventArgs)
			If Me.InvokeRequired Then
				Me.BeginInvoke(New EventHandler(AddressOf MainLookAndFeel_StyleChanged), sender, e)
				Return
			End If

			UserLookAndFeel.Default.Assign(CType(sender, UserLookAndFeel))
		End Sub

	End Class
End Namespace